bridges/YoutubeBridge.php

```diff
@@ -102,9 +136,105 @@ private function ytBridgeAddItem($vid, $title, $author, $desc, $time){
 		$item['uri'] = self::URI . 'watch?v=' . $vid;
 		$thumbnailUri = str_replace('/www.', '/img.', self::URI) . 'vi/' . $vid . '/0.jpg';
 		$item['content'] = '<a href="' . $item['uri'] . '"><img src="' . $thumbnailUri . '" /></a><br />' . $desc;
+		$this->ytBridgeAttachVideo($vid, $item);
 		$this->items[] = $item;
 	}
 
+	private function ytBridgeAttachVideo($vid, &$item) {
+		$attach_format = $this->getInput('attach_format');
+
+		if(empty($attach_format)) {
+			return;
+		}
+		$path = $this->ytBridgeDownloadVideo($vid, $attach_format);
+		if($path === null) {
+			return;
+		}
+
+		$path = realpath($path);
+		$mainDir = realpath(PATH_ROOT);
+
+		if(strpos($path, $mainDir) !== 0) {
+			error_log("Downloaded video in $path is not inside $mainDir");
+			return;
+		}
+
+		$urlPrefix = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] == 'on') ? 'https://' : 'http://';
```

Comment:
I copied this from `JsonFormat.php` where it’s used to construct the feed URL. I think this should move to a central location and the base URL (scheme, host, maybe a path prefix) needs to be configurable, e.g. for reverse proxy setups.

Reply:
Is it possible to have relative links? What does the spec say?